common/custom_error.go

```diff
@@ -0,0 +1,34 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+)
+
+type CustomError map[string]string
+
+func (e CustomError) Error() string {
+	return fmt.Sprintf("%s(%#v)", e["name"], e)
+}
+
+func (e CustomError) HTTPError(ctx context.Context) HTTPError {
+	httpStatusString := getOrDefault(e, "http_status", "500")
+	httpStatus, err := strconv.Atoi(httpStatusString)
+	if err != nil {
+		logEntry := GetLogEntryFromContext(ctx)
+		logEntry.Error(fmt.Sprintf("invalid http_status: %s for: %s", httpStatusString, e["name"]))
+		httpStatus = 500 //nolint: // TODO: use constant for internal server error
+	}
+	httpCode := getOrDefault(e, "http_code", "")
+	httpMessage := getOrDefault(e, "http_message", "")
+	return HTTPError{httpStatus, httpCode, httpMessage}
```

Comment:
Is there a reason we are making a copy of the `HTTPError` here? Could `httpMessage` be large? Would returning a pointer help with:
```Go
e := t.HTTPError(ctx)
httpError = &e
```

Reply:
>  Could httpMessage be large?

It's unlikely to be of any great length but yes, a pointer can be returned here.